pkgs/misc/mosquitto-go-auth/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, lib, buildGoPackage, fetchFromGitHub, pkgs }:
+
+buildGoPackage rec {
+  name = "mosquitto-go-auth-${version}";
+  version = "v0.2.0";
+  rev = "0.2.0";
+
+  goPackagePath = "github.com/iegomez/mosquitto-go-auth";
+
+  src = fetchFromGitHub {
+    inherit rev;
+    owner = "iegomez";
+    repo = "mosquitto-go-auth";
+    sha256 = "0wmi1w6nzl2nf0shrfb17n61ny625gpys03xlpx4h4sagvlwrsyn";
+  };
+
+  goDeps = ./deps.nix;
+
+  buildInputs = with pkgs; [
+    mosquitto
+  ];
+
+  outputs = [ "out" ];
+
+  dontRenameImports = true;
+  allowGoReference = true;
```

Comment:
Whether this will be needed or not if you'll switch to `buildGoModule`, could you at least explain in a comment why do these two are needed?